### PR TITLE
44 refactor geocoding

### PR DESCRIPTION
**Pull request by ivanoblomov**
Related to: #44 (item 1)

 ## Goal
Generalize `TestSites::Geocoder` to work with `CACLocations`.

 ## Approach
1. Refactor `TestSites::Geocoder#process` to accept location records
2. Test 1 with `CACLoation.all_from_api`.
3. Add coverage for 1 and 2.
4. Refactor `CACLoation.all_from_api` to use `ComponentizedUSAddress`.

### CODE REVIEW

**Comment by ivanoblomov** [comment]
Investigate this test-run output:
```
WARN -- : You are setting a key that conflicts with a built-in method Hashie::Mash#class defined in Kernel. This can cause unexpected behavior when accessing the key as a property. You can still access the key via the #[] method.
```

**Comment by ivanoblomov** [comment]
@neilberkman ready for review, but I'm not sure if the [above warning](https://github.com/Spark-Bio/covid-storepoint/pull/49#issuecomment-619270294) is significant.

File: lib/models/cac_location.rb [hunk 1]
```diff
@@ -67,6 +67,11 @@ def self.to_storepoint(locations)
     locations.map(&:to_storepoint)
   end
 
+  def address
```
Comment (neilberkman) [inline lib/models/cac_location.rb:70]: @ivanoblomov `address` is just street address, not the full address.

You should instantiate a `ComponentizedUSAddress` using the geocoder result for the full address and then call it for address / city / state / postcode. See [Listing](https://github.com/Spark-Bio/covid-storepoint/blob/master/lib/test_sites/listing.rb#L31-L45) for an example.

File: lib/models/cac_location.rb [hunk 2]
```diff
@@ -48,15 +48,30 @@ class CACLocation
...
   #
   # @return [Array] all CACLocations from the API
+  # rubocop:disable Metrics/MethodLength
   def self.all_from_api
-    TestSites::CAC.cac_data.map do |hash|
-      CACLocation.new(hash)
+    geocoder_results = TestSites::GeocoderResults.new.filtered
+    locations = []
+
+    TestSites::CAC.cac_data.each do |json|
```
Comment (neilberkman) [inline lib/models/cac_location.rb:61]: small nitpick on naming - it's a hash (actually a mash), not json

Comment (neilberkman) [reply lib/models/cac_location.rb:61]: also, why not use `map` rather than incrementally appending to an array within an `each`?

**Comment by ivanoblomov** [comment]
@neilberkman let me know if you have concerns with anything else.

**Comment by neilberkman** [comment]
Closed in favor of #54

**Pull request closed without merging**